test-case/stress-reboot.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+##
+## Case Name: stress test reboot
+## Preconditions:
+##    N/A
+## Description:
+##    run reboot for the stress test
+##    default duration is 10s
+##    default loop count is 3
+## Case step:
+##    1. check system status is correct
+##    2. wait for the Random value
+##    3. trigger for the reboot
+## Expect result:
+##    Test execute without report error in the LOG
+##
+
+source $(dirname ${BASH_SOURCE[0]})/../case-lib/lib.sh
+
+OPT_OPT_lst['l']='loop'     OPT_DESC_lst['l']='loop count'
+OPT_PARM_lst['l']=1         OPT_VALUE_lst['l']=5
+
+func_opt_parse_option $*
+
+func_lib_check_sudo
+func_lib_trigger_stress
+
+loop_count=${OPT_VALUE_lst['l']}
+
+if [ $loop_count -le 0 ]; then
+    mv $LOG_ROOT/cmd-orig.txt $LOG_ROOT/cmd.txt -f
+    exit 0
+fi
+# write the orig & status count to the status file
+echo $loop_count >> $STRESS_STATUS_LOG
+current=$(wc -l $STRESS_STATUS_LOG|awk '{print $1;}')
+count=$(head -n 1 $STRESS_STATUS_LOG|awk '{print $1;}')
+# because write record by desc
+current=$[ $count - $current + 1 ]
+
+dlogi "Round: $current/$count" | tee -a $STRESS_OUTPUT_LOG
+
+declare -a verify_lst
+verify_lst=(${verify_lst[*]} "verify-firmware-presence.sh")
+verify_lst=(${verify_lst[*]} "verify-kernel-module-load-probe.sh")
+verify_lst=(${verify_lst[*]} "verify-pcm-list.sh")
+verify_lst=(${verify_lst[*]} "verify-sof-firmware-load.sh")
+verify_lst=(${verify_lst[*]} "verify-tplg-binary.sh")
+
+# here will write 2 line into $STRESS_OUTPUT_LOG
+# because for the 1st round those information write by nohup command redirect
+# but after system reboot the nohup redirect is missing
+dlogi "Check status begin" | tee -a $STRESS_OUTPUT_LOG
+for i in ${verify_lst[*]}
+do
+    #$(dirname ${BASH_SOURCE[0]})/$i
+    if [ 0 -ne 0 ];then
+        # last line add failed keyword
+        sed -i '$s/$/ fail/' $STRESS_STATUS_LOG
+        dlogi "$i: fail" |tee -a $STRESS_OUTPUT_LOG
```

Comment:
Can you hijack the dlog* cmd to wrap the `|tee -a $STRESS_OUTPUT_LOG` when `STRESS_OUTPUT_LOG` is defined?

Reply:
that's the good idea, I will try it

Reply:
check with #5 , I don't think here need to modify  
because when run from remote, the `nohup` command will handle all output  
from the local it don't need to handle it  

here use `tee` command because after system reboot,
the `nohup` command is miss, Because  current script is run in `rc.local` which called by root permission